fda/registration.py

```diff
@@ -4,38 +4,60 @@
 functional data and related routines, in basis form as well in discretized form.
 
 """
+import enum
 
 import numpy
 import scipy.integrate
 
 
-def _check_extrapolation(fd, ext):
-    # Check how to extrapolate
-    # By default uses the value of the fd object
-    if ext == "default" or ext == 0:
-        ext = fd.default_extrapolation
+class ExtrapolationType(enum.Enum):
+    r"""Enum with extrapolation types.
 
-    if ext == "extrapolation" or ext == 1:
-        periodic = 1
-        extrapolation = False
+        Defines the extrapolation mode for elements outside the domain range.
 
-    elif ext == "periodic" or ext == 2:
-        extrapolation = 2
-        periodic = True
+        * default: uses the default method defined in the object to extrapolate.
+        * extrapolation: uses the extrapolated values directly.
+        * periodic: extends the domain range periodically.
+        * const: uses the boundary value.
+        * slice: avoids extrapolation restricting the domain.
+    """
+    default = 0
+    extrapolation = 1
+    periodic = 2
+    const = 3
+    slice = 4
 
-    elif ext == "const" or ext == 3:
-        extrapolation = 3
-        periodic = True
+    @classmethod
+    def parse(cls, arg, fd=None):
+        r"""Parse an ExtrapolationType from string, int or ExtrapolationType.
 
-    elif ext == "slice" or ext == 4:
-        extrapolation = 1
-        periodic = False
+            Args:
+                arg (str, int or ExtrapolationType): Argument to parse. Can
+                    be the number of the extrapolation type or an string with
+                    the name.
+                fd (:obj:`FDataBasis`, optional): Functional data object with
+                    a default extrapolation type defined.
+            Return:
+                :obj:`ExtrapolationType`: corresponding type parsed.
+        """
 
-    else:
-        raise ValueError("Incorrect value of ext, should be 'default', 'basis'"
-                         ", 'periodic' or 'slice'.")
+        if isinstance(arg, ExtrapolationType): arg = arg.value
+        elif isinstance(arg, str): arg = arg.lower()
+
+        for ext in ExtrapolationType:
+
+            if arg == ext.value or arg == ext.name:
+
+                if ext.value == ExtrapolationType.default.value:
```

Comment:
I think it is easier to write:
`if ext is ExtrapolationType.default:`

Reply:
Fixed in 9c17e4d.